recipes/LJSpeech/quantization/hparams/train_discrete_ssl.yaml

```diff
@@ -1,78 +1,102 @@
-################################
-# Recipe for Training K-Means Clustering on LJSpeech Data
-# Using Self-Supervised Model-Based Representations
-#
-# It is used for creating discrete audio representations from LJSpeech data.
-#
-# Author: Pooneh Mousavi (2023)
-################################
-# Seed needs to be set at top of yaml, before objects with parameters are made
-seed: 1986
-__set_seed: !apply:speechbrain.utils.seed_everything [!ref <seed>]
-output_folder: !ref results/LJSpeech/clustering/hubert/<seed>
-save_folder: !ref <output_folder>/save
+# ###########################################################################################
+# Model: K-means applied to SSL model
+# Authors: Luca Della Libera 2024
+# ###########################################################################################
 
-# Data files
-data_folder: !PLACEHOLDER # e,g./path/to/LJSpeech-1.1
+experiment_name: wav2vec2_K1000_L7
 
-train_json: !ref <save_folder>/train.json
+# Seed needs to be set at top of YAML
+seed: 1986
+__set_seed: !apply:speechbrain.utils.seed_everything [!ref <seed>]
 
-splits: ["train"]
-split_ratio: [80]
+# Data preparation
+data_folder: !PLACEHOLDER
+splits: [train, valid, test]
+split_ratio: [80, 10, 10]
 skip_prep: False
-sample_rate: 16000
+train_json: !ref <output_folder>/train.json
+valid_json: !ref <output_folder>/valid.json
+test_json: !ref <output_folder>/test.json
 
-# model_config
-# ssl_model_type: hubert, wavlm, wav2vec2
-# ssl_hub: facebook/hubert-large-ll60k, microsoft/wavlm-large,  facebook/wav2vec2-large
-ssl_model_type: hubert # hubert, wavml or wav2vec2
-ssl_hub: facebook/hubert-large-ll60k
-freeze_feature_extractor: True
-freeze_ssl: True
-ssl_folder: !ref <save_folder>/hubert_checkpoint
-ssl_layer_num: 7
-batch_size: 128 # batch_size for loading and extracting features. It is different from kmeans_batch_size.
-checkpoint_interval: 100
+# Output folders
+output_folder: !ref results/<experiment_name>/<seed>
+save_folder: !ref <output_folder>/save
+cache_folder: !name:huggingface_hub.constants.HUGGINGFACE_HUB_CACHE
 
-# Dataloader options
-train_dataloader_opts:
-    batch_size: !ref <batch_size>
-    drop_last: True
+# Preprocessing parameters
+# For some reason the duration is not available in the manifest file...
```

Comment:
abt this, can't we simply modify the prepare recipe to have an extra duration column ?

Reply:
Done